x-pack/plugins/fleet/dev_docs/developing_kibana_and_fleet_server.md

```diff
@@ -0,0 +1,278 @@
+# Developing Kibana and Fleet Server simulatanously
+
+Many times, a contributor to Fleet will only need to make changes to [Fleet Server](https://github.com/elastic/fleet-server) or [Kibana](https://github.com/elastic/kibana) - not both. But, there are times when end-to-end changes across both componenents are necessary. To facilitate this, we've created a guide to help you get up and running with a local development environment that includes both Kibana and Fleet Server. This is a more involved process than setting up either component on its own.
+
+This guide seeks to get you up and running with the following stack components for local development:
+
+- Kibana (from source)
+- Elasticsearch (from a snapshot)
+- Fleet Server (from source, in standalone mode)
+
+Getting this development environment up and running will allow you to make changes to both Kibana and Fleet Server simultaneously to develop and test features end-to-end.
+
+Before continuing, please review the developer documentation for both Fleet Server and Kibana. Make sure your local workstation can run each service independently. This means setting up Go, Node.js, Yarn, resolving dependencies, etc.
+
+- https://github.com/elastic/fleet-server?tab=readme-ov-file#development
+- https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md
+
+## Start up Kibana and Elasticsearch
+
+1. Get a local Kibana checkout up and running by following [the developer guide](https://docs.elastic.dev/kibana-dev-docs/getting-started/setup-dev-env) (internal Elastic employee link for now). Run the commands below from your `kibana` directory
+2. Run a local Elasticsearch server from the latest snapshot build by running:
+
+```bash
+# Enable SSL + set a data path to prevent blowing away your Elasticsearch data between server restarts
+$ yarn es snapshot --license trial --ssl -E path.data=/tmp/es-data
```

Comment:
nit: I've been using local es without ssl, it works with fleet-server as well

Reply:
It's probably worth documenting this both with SSL and without. I'll take a pass at that. The docker stuff will need to be updated to account for both options as well.

Reply:
I've added separate HTTP/HTTPS instructions to address this. HTTP is indeed a lot easier and works fine for most development use cases.